### PR TITLE
Target logits bug fix

### PR DESCRIPTION
**Pull request by ViktorDomazetoski**
When using a batch size above 1, the returned target logits for the entire batch are equal to the first element of the batch. The fix enumerates through the predictions to get the logit for the corresponding element of the batch.

### CODE REVIEW

**Comment by stefanistrate** [comment]
Thanks for the fix, @ViktorDomazetoski!

Once you sign the CLA mentioned above and address the formatting issues identified by our automatic checks (i.e. https://github.com/google/cameratrapai/actions/runs/17946293131/job/51082087160?pr=45), we're happy to approve your change.

If you're feeling even more adventurous, since you uncovered a bug for which we didn't test properly, feel free to add a test in `classifier_test.py` to verify that using `target_idx` works correctly with and without batches.

**Review by stefanistrate — APPROVED** [review]
Thank you for your contribution, @ViktorDomazetoski! Looking great.
